tests/vxlan/test_vxlan.py

```
@@ -0,0 +1,653 @@
+            po_config_list.append(config)
+            for member in get_ethernet_ports([intf], minigraph_data):
+                ret_list[member] = vnet_list[count]
+
+    full_config_list = []
+    if intf_config_list:
+        full_config_list.append(
+            '''"INTERFACE": {\n''' + ",\n".join(intf_config_list) + '''}''')
+    if po_config_list:
+        full_config_list.append(
+            '''"PORTCHANNEL_INTERFACE": {\n''' + ",\n".join(po_config_list) + '''}''')
+
+    full_config = '''{\n''' + ",\n".join(full_config_list) + '''}'''
+    apply_config_in_dut(duthost, full_config, "vnet_intf")
+    return ret_list
+
+def configure_vxlan_switch(duthost, vxlan_port=4789, dutmac=None):
+    if dutmac == None:
+        #dutmac = duthost.facts['router_mac']
+        dutmac = "aa:bb:cc:dd:ee:ff"
+
+    switch_config = '''
+[
+        {{
+                "SWITCH_TABLE:switch": {{
+                        "vxlan_port": "{}",
+                        "vxlan_router_mac": "{}"
+                }},
+                "OP": "SET"
+        }}
+]
+'''.format(vxlan_port, dutmac)
+    apply_config_in_swss(duthost, switch_config, "vnet_switch")
+
+def apply_config_in_swss(duthost, config, name="swss_"):
+    if Constants['DEBUG']:
+        filename = name + ".json"
+    else:
+        filename = name + "-" + str(time.time()) + ".json"
+
+    duthost.copy(content=config, dest="/tmp/{}".format(filename))
+    result = duthost.shell('docker exec -i swss swssconfig /dev/stdin < /tmp/{}'.format(filename))
+    if not Constants['KEEP_TEMP_FILES']:
+        duthost.shell("rm /tmp/{}".format(filename))
+    time.sleep(1)
+
+def generate_random_mac():
+    return "02:00:00:%02x:%02x:%02x" % (random.randint(0, 255),
+                                        random.randint(0, 255),
+                                        random.randint(0, 255))
+
+def get_list_of_nexthops(number, af, prefix=None):
+    if prefix is None:
+        prefix = random.randint(0, 220)
```

Let keep the prefix fixed to avoid a lot of randomness

---

I will modify the code to keep the prefix to a user defined parameter, and default to a fixed number.

---

I implemented the prefix as a hardcoded value in the script. The user given parameters can cause conflict with the routes in the topology and cause script failures.